### ISSUE TITLE
Pretrained VGG-19 Model

### ISSUE THREAD

**Issue by junting** [pytorch/vision#433]
Hi,

I have been using the pretrained ResNet model which works fine, however when I switched to the pretrained VGG-19 at the master branch, I found that the output of the the pytorch verison is very different from the caffe version.  The first predicted class for this [https://github.com/BVLC/caffe/blob/master/examples/images/cat.jpg](url) is 644 which is wrong. 
The preprocessing that I used is what was suggested by @soumith   here [https://discuss.pytorch.org/t/whats-the-range-of-the-input-value-desired-to-use-pretrained-resnet152-and-vgg19/1683](url) 

`import torchvision.models as models`
`from torchvision import transforms as trn`
`import skimage.io`
`import skimage`

`preprocess = trn.Compose([
    trn.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])
])`

`vgg19 = models.vgg19()`
`vgg19.load_state_dict(torch.load(os.path.join(model_path, 'vgg19-dcbb9e9d.pth')))`
`vgg19.cuda()`
`vgg19.eval()`

`I = skimage.io.imread(image_path)`

`I = crop_image(I)`
`I = torch.from_numpy(I.transpose([2, 0, 1])).cuda()`
`I = Variable(preprocess(I), volatile=True)`
`output = vgg19.forward(I.unsqueeze(0))`
`output = F.softmax(output, dim=1)`

Should the preprocessing be different from the one suggested by @soumith ?
Thanks,

**Comment by fmassa** [comment]
The preprocessing for vgg is the same as for resnet.

What I think is happening is that `skimage.io.imread` returns a numpy array in the range 0-255, and I see that you are not using the `ToTensor()` transform (which converts the PIL Image to be in the 0-1 range).
Also, make sure that your image has float type before you perform `Normalize`, if not you'll have wrong results as well.

I'm closing the issue because I believe this is the root cause of the error.
Please let me know if you still face the same issues.

**Comment by junting** [comment]
Hi,

Thanks for your quick response, the problem comes from image image which should be between 0-1. 
Thank you,
